R/pkg/R/sparkR.R

```diff
@@ -365,6 +365,10 @@ sparkR.session <- function(
     }
     overrideEnvs(sparkConfigMap, paramMap)
   }
+  if (nzchar(master)) {
+    assign("spark.master", master, envir = sparkConfigMap)
```

Comment:
It is passed to `createSparkContext`, but it is too late as it should be passed to JVM when it is started. Another approach is to duplicate the code in SparkSubmit.scala to add sparkr.zip to archive.